src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -149,6 +241,102 @@ void CascadiaSettings::SaveAll() const
     _WriteSettings(serializedString);
 }
 
+// Method Description:
+// - Finds all the dynamic profiles we've generated that _don't_ exist in the
+//   user's settings. Generates a minimal blob of json for them, and inserts
+//   them into the user's settings at the end of the list of profiles.
+// - Does not reformat the user's settings file.
+// - Does not write the file! Only modifies in-place the _userSettingsString
+//   member. Callers should make sure to call
+//   _WriteSettings(_userSettingsString) to make sure to persist these changes!
+// - Assumes that the `profiles` object is at an indentation of 4 spaces, and
+//   therefore each profile should be indented 8 spaces. If the user's settings
+//   have a different indentation, we'll still insert valid json, it'll just be
+//   indented incorrectly.
+// Arguments:
+// - <none>
+// Return Value:
+// - true iff we've made changes to the _userSettingsString that should be persisted.
+bool CascadiaSettings::_AppendDynamicProfilesToUserSettings()
+{
+    // - Find the set of profiles that weren't either in the default profiles or
+    //   in the user profiles. TODO: Do this in not O(N^2)
+    // - For each of those profiles,
+    //   * Diff them from the default profile
+    //   * Serialize that diff
+    //   * Insert that diff to the end of the list of profiles.
+
+    const Profile defaultProfile{};
+
+    Json::StreamWriterBuilder wbuilder;
+    // Use 4 spaces to indent instead of \t
+    wbuilder.settings_["indentation"] = "    ";
+
+    auto isInJsonObj = [](const auto& profile, const auto& json) {
+        for (auto profileJson : _GetProfilesJsonObject(json))
+        {
+            if (profileJson.isObject())
+            {
+                if (profile.ShouldBeLayered(profileJson))
+                {
+                    return true;
+                }
+            }
+        }
+        return false;
+    };
+
+    // Get the index in the user settings string of the _last_ profile.
+    // We want to start inserting profiles immediately following the last profile.
+    const auto userProfilesObj = _GetProfilesJsonObject(_userSettings);
+    const auto numProfiles = userProfilesObj.size();
+    const auto lastProfile = userProfilesObj[numProfiles - 1];
+    size_t currentInsertIndex = lastProfile.getOffsetLimit();
+
+    bool changedFile = false;
+
+    for (const auto& profile : _profiles)
+    {
+        // Skip profiles that are in the user settings or the default settings.
+        if (isInJsonObj(profile, _userSettings) || isInJsonObj(profile, _defaultSettings))
+        {
+            continue;
+        }
+
+        // Generate a diff for the profile, that contains the minimal set of
+        // changes to re-create this profile.
+        const auto diff = profile.DiffToJson(defaultProfile);
+        auto profileSerialization = Json::writeString(wbuilder, diff);
+
+        // Add 8 spaces to the start of each line
+        profileSerialization.insert(0, "        ");

```

Comment:
For the "8 spaces" - can this be a constant?  This way you can change this in one place.